themes/wilson_theme_starterkit/templates/paragraph/paragraph--anchor-point.html.twig

```diff
@@ -0,0 +1,41 @@
+{#
+/**
+ * @file
+ * Default theme implementation to display an Anchor point paragraph.
+ *
+ * Available variables:
+ * - paragraph: Full paragraph entity.
+ *   Only method names starting with "get", "has", or "is" and a few common
+ *   methods such as "id", "label", and "bundle" are available. For example:
+ *   - paragraph.getCreatedTime() will return the paragraph creation timestamp.
+ *   - paragraph.id(): The paragraph ID.
+ *   - paragraph.bundle(): The type of the paragraph, for example, "image" or "text".
+ *   - paragraph.getOwnerId(): The user ID of the paragraph author.
+ *   See Drupal\paragraphs\Entity\Paragraph for a full list of public properties
+ *   and methods for the paragraph object.
+ * - content: All paragraph items. Use {{ content }} to print them all,
+ *   or print a subset such as {{ content.field_example }}. Use
+ *   {{ content|without('field_example') }} to temporarily suppress the printing
+ *   of a given child element.
+ * - attributes: HTML attributes for the containing element.
+ *   The attributes.class element may contain one or more of the following
+ *   classes:
+ *   - paragraphs: The current template type (also known as a "theming hook").
+ *   - paragraphs--type-[type]: The current paragraphs type. For example, if the paragraph is an
+ *     "Image" it would result in "paragraphs--type--image". Note that the machine
+ *     name will often be in a short form of the human readable label.
+ *   - paragraphs--view-mode--[view_mode]: The View Mode of the paragraph; for example, a
+ *     preview would result in: "paragraphs--view-mode--preview", and
+ *     default: "paragraphs--view-mode--default".
+ * - view_mode: View mode; for example, "preview" or "full".
+ * - logged_in: Flag for authenticated user status. Will be true when the
+ *   current user is a logged-in member.
+ * - is_admin: Flag for admin user status. Will be true when the current user
+ *   is an administrator.
+ *
+ * @see template_preprocess_paragraph()
+ *
+ * @ingroup themeable
+ */
+#}
+<a class="anchor-point" id="{{ content.anchor_point_id }}"></a>
```

Comment:
Does the anchor point need to be a link? If this is the part of the page that would get scrolled to, could this just be a div with an ID? I'm wondering what the accessibility implications are for having an empty link on the page, which can be tabbed to?

I know that Drupal core uses a named link with a negative tabindex e.g.

```
<a name="main-content" tabindex="-1"></a>
```

However I've heard that named anchors are no longer conforming to HTML5 https://webaim.org/techniques/skipnav/#visible

Reply:
I might have missed something here, but as far as I am aware, an anchor tag is only a link (and interactive) if it has a href. You can't tab to a plain anchor tag. Accessibility wise, it should still have some content, even if it is visibly hidden.

I think the tabindex on Drupal's content anchor isn't taking it out of the tab order (it was never there) but making it click/js focusable (for reasons I don't understand 😄 )

Regardless of this your suggestion of using an ID on another element is valid. That's actually how they do in-page nav on the WCAG guidelines page https://www.w3.org/TR/WCAG21/#background-on-wcag-2

Reply:
Because I wanted to make the anchor point a paragraph in it's own right - so they can be dragged and placed in the page editor between other components - I can't simply add the ID a wrapper around content. I had assumed using an anchor tag with an ID would be the more modern equivalent of doing a named anchor.

Do we see any problems with doing this?

```
<div class="anchor-point" id="{{ content.anchor_point_id }}"></div>
```

Reply:
I would prefer the `a` over a `div` then as that is the intended function. I would also put the anchor point label inside if there is one - visually hidden if you don't want it to show.

Reply:
I've updated with the anchor point label and marked it for screen readers only.